The Secplugs Java kit

benignbala opened the pull request:
This provides a REST client for Java and JVM in general. This Groovy implementation
can be leveraged by any JVM language.

benignbala committed:
The Secplugs Java kit

This provides a REST client for Java and JVM in general. This Groovy implementation
can be leveraged by any JVM language.

benignbala committed:
Add support for submitting email body for scan through proxy scanner

benignbala committed:
Add support for scancontext

The java-kit now allows the user to add scan context to the scan submissions.
This will make the reports ook more meaningful with the name of the scan object
reported as part of the scan context.

benignbala committed:
Add scancontext query param only if it is non-zero size

Adding an empty scancontext isn't useful. Send it only if it is non-zero

TheStigAtSecPlugs reviewed (commented):
Question on why we'd want to expose the vendor choice in the API - that is something should be configured in the portal and the plugin should be agnostic to. Was there a use case?

libretools commented:
> Question on why we'd want to expose the vendor choice in the API - that is something should be configured in the portal and the plugin should be agnostic to. Was there a use case?

The use case is to allow the user to choose a specific vendor if they need it.

TheStigAtSecPlugs commented:
> > Question on why we'd want to expose the vendor choice in the API - that is something should be configured in the portal and the plugin should be agnostic to. Was there a use case?
> 
> The use case is to allow the user to choose a specific vendor if they need it.

Can we make the default no vendor (rather than hybrid analysis)? 

The only real use case I can think of for exposing the vendor is so that people can use mock_vendor or mock_vendor_asynch when testing there plugin. 

We want to enforce good behaviour and support paradigm (there is so much we can do if the vendor choice is abstracted to the back end).

benignbala committed:
Do not send the x-vendorcfg header if it is empty

Instead of defaulting to some sepecific vendorid, use the one supplied to the API

benignbala committed:
Add support for URL scan

Adding the wrapper jar for bootstrapping the build and added URL scan support.
